chalice/app.py

```diff
@@ -765,3 +779,82 @@ class AuthRoute(object):
     def __init__(self, path, methods):
         self.path = path
         self.methods = methods
+
+
+class EventSource(object):
```

Comment:
Makes sense. I like the idea of having `EventSource` be the base class and `CloudWatchEventSource` subclass from it to include any of the cloudwatch event specific logic.